07-http_gordle/final/internal/handlers/guess/guess.go

```diff
@@ -0,0 +1,114 @@
+package guess
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+
+	"learngo-pockets/httpgordle/api"
+	"learngo-pockets/httpgordle/internal/domain"
+	"learngo-pockets/httpgordle/internal/gordle"
+	"learngo-pockets/httpgordle/internal/handlers/apiconversion"
+	"learngo-pockets/httpgordle/internal/repository"
+)
+
+type gameGuesser interface {
+	Find(domain.GameID) (domain.Game, error)
+	Update(domain.GameID, domain.Game) error
+}
+
+// Handler returns a handler for guess requests.
+func Handler(repo gameGuesser) http.HandlerFunc {
+	return func(writer http.ResponseWriter, request *http.Request) {
+		// Read the Game ID from the query parameters.
+		id := chi.URLParam(request, api.GameID)
+		if id == "" {
+			http.Error(writer, "missing the id of the game", http.StatusNotFound)
+		}
+
+		// Read the request, containing the guess, from the body of the input.
+		r := api.GuessRequest{}
+		err := json.NewDecoder(request.Body).Decode(&r)
+		if err != nil {
+			http.Error(writer, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		game, err := play(repo, domain.GameID(id), r.Guess)
+		if err != nil {
+			switch {
+			case errors.Is(err, repository.ErrNotFound):
+				http.Error(writer, err.Error(), http.StatusNotFound)
+			case errors.Is(err, gordle.ErrInvalidGuess):
+				http.Error(writer, err.Error(), http.StatusBadRequest)
+			case errors.Is(err, domain.ErrGameOver):
+				http.Error(writer, err.Error(), http.StatusForbidden)
+			default:
+				http.Error(writer, err.Error(), http.StatusInternalServerError)
+			}
+			return
+		}
+
+		apiGame := apiconversion.ToAPIResponse(game)
+
+		writer.WriteHeader(http.StatusAccepted)
```

Comment:
Writing header should happen after setting the headers. Setting headers after `WriteHeader` has no affect.

`http.StatusAccepted` is not correct here either, because it implies that the operation has not been completed yet.